Assets/Scripts/Controllers/AudioManager.cs

```diff
@@ -0,0 +1,130 @@
+        return "{" + string.Join(",", dictionary.Select(kv => kv.Key + "=" + kv.Value.length).ToArray()) + "}";
+    }
+
+    /// <summary>
+    /// Gets an AudioCLip from the specified category, with the specified name.
+    /// Throws a LogWarning and returns an Error sound if the AudioClip does not exist.
+    /// </summary>
+    /// <param name="categoryName">
+    /// The category that the AudioClip is in. Usually the same as the 
+    /// directory that the audio file was load from.
+    /// </param>
+    /// <param name="audioName">The name of the AudioClip.</param>
+    /// <returns>AudioClip form the specified category with the specified name.</returns>
+    public static AudioClip GetAudio(string categoryName, string audioName)
+    {
+        AudioClip clip = new AudioClip();
+
+        string audioNameAndCategory = categoryName + "/" + audioName + ".ogg";
+
+        if (audioClips.ContainsKey(audioNameAndCategory))
+        {
+            clip = audioClips[audioNameAndCategory];
+        }
+        else
+        {
+            try
+            {
+                Debug.LogWarning("No audio available called: " + audioNameAndCategory);
+                clip = audioClips["Sound/Error.ogg"];
+            }
+            catch
+            {
+                throw new FileNotFoundException("Sound/Error.ogg not found");
+            }
+
+        }
+
+        return clip;
+    }
+
+    /// <summary>
+    /// Loads all the audio files from the specified directory.
+    /// </summary>
+    /// <param name="directoryPath">The path of the directory you want to load the audio files from.</param>
+    public static void LoadAudioFiles(string directoryPath)
+    {
+        string[] subDirectories = Directory.GetDirectories(directoryPath);
+        foreach (string subDirectory in subDirectories)
+        {
+            LoadAudioFiles(subDirectory);
+        }
+
+        string[] filesInDir = Directory.GetFiles(directoryPath);
+        IEnumerable<WWW> audioFiles = LoadAudioFile(filesInDir, directoryPath);
+        IEnumerator<WWW> e = audioFiles.GetEnumerator();
+        while (e.MoveNext())
+        {
+            Debug.Log("Type: " + e.Current.GetType());
+        }
+    }
+
+    private static IEnumerable<WWW> LoadAudioFile(string[] filesInDir, string directoryPath)
+    {
+        foreach (string file in filesInDir)
+        {
+            string audioCategory = new DirectoryInfo(directoryPath).Name;
+            string filePath = new FileInfo(file).FullName;
+
+            if (filePath.Contains(".xml") || filePath.Contains(".meta") || filePath.Contains(".db"))
+            {
+                continue;
+            }
+
+            WWW www = new WWW(@"file://" + filePath);
+
+            Debug.Log("before - " + www.error);
+
+            yield return www;
+
+            Debug.Log("After - " + www.error);
+
+            AudioClip clip = www.GetAudioClip(false, false);
+
+            Debug.Log(clip.name + "Downloaded");
```

Comment:
`clip.name` doesn't seem to be populated in my testing?? But one informative log message per file is all you need. Would prefer something like
`Debug.ULogChannel("AudioManager", "Audio file loaded: '{0}'.", clip.name)`, assuming `clip.name` was set properly.